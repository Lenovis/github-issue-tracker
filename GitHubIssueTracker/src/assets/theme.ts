import { NativeStackNavigationOptions } from '@react-navigation/native-stack';

const HexAlpha: { [key: number]: string } = {
  0.9: 'E6',
  0.8: 'CC',
  0.7: 'B3',
  0.6: '99',
  0.5: '80',
  0.4: '66',
  0.3: '4D',
  0.2: '33',
  0.1: '1A',
};

export type ThemeType = typeof theme;

export type ThemeColors = typeof theme.colors;

export const theme = {
  colors: {
    backgroundColor: '#0e1120',
    black: '#000000',
    borderColor: '#cad1db',
    buttonGreen: '#438440',
    grey: '#94988e',
    blueGrey: '#171b21',
    greyAlpha: (alpha = 1) => `rgba(34, 34, 34, ${alpha})`, //is it in use?
    opacity: (hex: string, alpha: number) => `${hex}${HexAlpha[alpha]}`,
    white: '#FFFFFF',
    lightGrey: '#D3D3D3',
    purple: '#9a76ef',
    red: '#f24141',
  },
};

export const navigationBarTheme: NativeStackNavigationOptions = {
  headerStyle: { backgroundColor: theme.colors.backgroundColor },
  headerTitleStyle: { color: theme.colors.white },
};
