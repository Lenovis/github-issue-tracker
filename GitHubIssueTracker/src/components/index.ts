//Butons
export * from './buttons/DefaultButton';
export * from './buttons/BackButton';
export * from './buttons/NextButton';
//Texts
export * from './texts/HeaderText';
//Icons
export * from './Icons';
//IssueItems
export * from './IssuesComponents/listItem';
export * from './IssuesComponents/issuesFilterButton';
export * from './IssuesComponents/issueListView';
