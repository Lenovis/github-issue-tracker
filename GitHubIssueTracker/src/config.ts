export const config = {
  BASE_URL: 'https://api.github.com/',
};
