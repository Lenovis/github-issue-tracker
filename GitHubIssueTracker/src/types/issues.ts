export type Issue = {
  id: number;
};

export type IssueRequest = {
  owner: string;
  repo: string;
  page?: number;
};
