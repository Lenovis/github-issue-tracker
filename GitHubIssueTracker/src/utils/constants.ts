export enum IssuesState {
  open = 'open',
  closed = 'closed',
}
