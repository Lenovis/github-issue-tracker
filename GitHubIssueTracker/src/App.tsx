/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the TypeScript template
 * https://github.com/react-native-community/react-native-template-typescript
 *
 * @format
 */
import React from 'react';
import { StatusBar } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import styled, { ThemeProvider } from 'styled-components';
import { theme as importedTheme } from './assets/theme';
import Navigator from './routes/Navigator';
import { Provider } from 'react-redux';
import { configStore } from './state/store';

const { store } = configStore();
const App = () => {
  return (
    <Provider store={store}>
      <ThemeProvider theme={importedTheme}>
        <Container>
          <StatusBar
            barStyle="light-content"
            backgroundColor={importedTheme.colors.backgroundColor}
          />
          <Navigator />
        </Container>
      </ThemeProvider>
    </Provider>
  );
};

export default App;

const Container = styled(SafeAreaView)`
  background-color: ${({ theme }) => theme.colors.backgroundColor};
  flex: 1;
`;
