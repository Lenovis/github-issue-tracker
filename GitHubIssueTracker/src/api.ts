import axios from 'axios';
import { config } from './config';
import { Issue } from './types';

type Status = {
  status: boolean;
};

export default class Api {
  static get = async <T>(endpoint: string) => {
    const url = config.BASE_URL + endpoint;

    return axios.get<T>(url);
  };

  static getRepoIssues = async (owner: string, repo: string, page = 1) =>
    Api.get<Issue[] & Status>(`repos/${owner}/${repo}/issues?page=${page}`);
}
