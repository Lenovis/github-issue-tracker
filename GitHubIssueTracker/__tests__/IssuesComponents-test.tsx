import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {
  CheckCircleIcon,
  DotCircleIcon,
  IssueListItemImage,
  MergedRequestIcon,
  PullRequestIcon,
} from '../src/components';
import 'jest-styled-components';
import { IssuesState } from '../src/utils';
import { IssuePullRequest } from '../src/types';
import { theme } from '../src/assets/theme';
import {
  FooterComponentView,
  IssueList,
  ListPagination,
} from '../src/components/IssuesComponents/issueListView';
import * as reactRedux from 'react-redux';
import { cleanup } from '@testing-library/react-native';
import { View } from 'react-native';

afterEach(cleanup);

//---issueListItemicon
test('renders correctly issues list item icon', () => {
  const tree = renderer.create(<IssueListItemImage />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly issues list item icon with state: open', () => {
  const tree = renderer
    .create(<IssueListItemImage state={IssuesState.open} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly issues list item icon with state: open, draft: true', () => {
  const tree = renderer
    .create(<IssueListItemImage state={IssuesState.open} draft={true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly issues list item icon with state: open, pull request', () => {
  const tree = renderer
    .create(<IssueListItemImage state={IssuesState.open} pull_request={{}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly issues list item icon with state: closed', () => {
  const tree = renderer
    .create(<IssueListItemImage state={IssuesState.closed} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly issues list item icon with state: closed, pull request', () => {
  const tree = renderer
    .create(<IssueListItemImage state={IssuesState.closed} pull_request={{}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly issues list item icon with state: closed, pull request merged at', () => {
  const tree = renderer
    .create(
      <IssueListItemImage
        state={IssuesState.closed}
        pull_request={{ merged_at: new Date() }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('given state open returns <DotCircleIcon />', () => {
  expect(IssueListItemImage({ state: IssuesState.open })).toStrictEqual(
    <DotCircleIcon />,
  );
});

test('given state open and draft true returns <PullRequestIcon color={theme.colors.grey} />', () => {
  expect(
    IssueListItemImage({
      state: IssuesState.open,
      draft: true,
    }),
  ).toStrictEqual(<PullRequestIcon color={theme.colors.grey} />);
});

test('given state open and header returns <PullRequestIcon />', () => {
  const IssuePullRequestWithoutDate = {} as IssuePullRequest;
  expect(
    IssueListItemImage({
      state: IssuesState.open,
      pull_request: IssuePullRequestWithoutDate,
    }),
  ).toStrictEqual(<PullRequestIcon />);
});

test('given state closed returns <CheckCircleIcon />', () => {
  expect(IssueListItemImage({ state: IssuesState.closed })).toStrictEqual(
    <CheckCircleIcon />,
  );
});

test('given state closed and header wihout merged time returns <PullRequestIcon color={theme.colors.red} />', () => {
  const IssuePullRequestWithoutDate = {} as IssuePullRequest;
  expect(
    IssueListItemImage({
      state: IssuesState.closed,
      pull_request: IssuePullRequestWithoutDate,
    }),
  ).toStrictEqual(<PullRequestIcon color={theme.colors.red} />);
});

test('given state closed and header with merged time returns <MergedRequestIcon />', () => {
  const IssuePullRequestWithtDate = {
    merged_at: new Date(),
  } as IssuePullRequest;
  expect(
    IssueListItemImage({
      state: IssuesState.closed,
      pull_request: IssuePullRequestWithtDate,
    }),
  ).toStrictEqual(<MergedRequestIcon />);
});

//---issueListView

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

describe('renders correctly gettingData: true', () => {
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');
  const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');

  beforeEach(() => {
    useDispatchMock.mockImplementation();
    useSelectorMock.mockImplementation(selector => selector(mockStore));
  });
  afterEach(() => {
    useDispatchMock.mockClear();
    useSelectorMock.mockClear();
  });

  const mockStore = {
    ui: { repoIssuesOnSync: true },
  };

  it('issueList(issues: [])', () => {
    const tree = renderer.create(IssueList([])).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('issueList(issues: [{issue}])', () => {
    const tree = renderer
      .create(
        IssueList([
          {
            id: 1,
            state: IssuesState.open,
            user: {},
            pull_request: {
              merged_at: new Date(),
            },
          },
        ]),
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('renders correctly gettingData: false', () => {
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');
  const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');

  beforeEach(() => {
    useDispatchMock.mockImplementation();
    useSelectorMock.mockImplementation(selector => selector(mockStore));
  });
  afterEach(() => {
    useDispatchMock.mockClear();
    useSelectorMock.mockClear();
  });

  const mockStore = {
    ui: { repoIssuesOnSync: false },
  };

  it('issueList(issues: [])', () => {
    const tree = renderer.create(IssueList([])).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('issueList(issues: [{issue}])', () => {
    const tree = renderer
      .create(
        IssueList([
          {
            id: 1,
            state: IssuesState.open,
            user: {},
            pull_request: {
              merged_at: new Date(),
            },
          },
        ]),
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

test('renders correctly <FooterComponentView hasPrevPage={true} hasNextPage={true} issuesCurrentPage={1}/>', () => {
  const tree = renderer
    .create(
      <FooterComponentView
        hasNextPage={true}
        hasPrevPage={true}
        issuesCurrentPage={1}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly <FooterComponentView hasPrevPage={false} hasNextPage={false} issuesCurrentPage={1}/>', () => {
  const tree = renderer
    .create(
      <FooterComponentView
        hasNextPage={false}
        hasPrevPage={false}
        issuesCurrentPage={1}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly <FooterComponentView hasPrevPage={true} hasNextPage={false} issuesCurrentPage={1}/>', () => {
  const tree = renderer
    .create(
      <FooterComponentView
        hasNextPage={true}
        hasPrevPage={false}
        issuesCurrentPage={1}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly <FooterComponentView hasPrevPage={false} hasNextPage={true} issuesCurrentPage={1}/>', () => {
  const tree = renderer
    .create(
      <FooterComponentView
        hasNextPage={false}
        hasPrevPage={true}
        issuesCurrentPage={1}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
