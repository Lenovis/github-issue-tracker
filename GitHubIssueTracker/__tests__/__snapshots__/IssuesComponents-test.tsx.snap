// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly <FooterComponentView hasPrevPage={false} hasNextPage={false} issuesCurrentPage={1}/> 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginBottom": 0,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontWeight": "bold",
        },
      ]
    }
  >
    1 page
  </Text>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
</View>
`;

exports[`renders correctly <FooterComponentView hasPrevPage={false} hasNextPage={true} issuesCurrentPage={1}/> 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginBottom": 0,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 12,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontWeight": "bold",
        },
      ]
    }
  >
    1 page
  </Text>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
</View>
`;

exports[`renders correctly <FooterComponentView hasPrevPage={true} hasNextPage={false} issuesCurrentPage={1}/> 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginBottom": 0,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontWeight": "bold",
        },
      ]
    }
  >
    1 page
  </Text>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 12,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly <FooterComponentView hasPrevPage={true} hasNextPage={true} issuesCurrentPage={1}/> 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginBottom": 0,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 12,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontWeight": "bold",
        },
      ]
    }
  >
    1 page
  </Text>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 12,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly gettingData: false issueList(issues: []) 1`] = `<View />`;

exports[`renders correctly gettingData: false issueList(issues: [{issue}]) 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#FFFFFF",
          "flexDirection": "row",
          "minHeight": 40,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "marginRight": 10,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#438440",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "marginBottom": 5,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontWeight": "bold",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#D3D3D3",
              },
            ]
          }
        >
          #undefined opened less than a minute ago by undefined
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginBottom": 0,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontWeight": "bold",
          },
        ]
      }
    >
      undefined page
    </Text>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`renders correctly gettingData: true issueList(issues: []) 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "marginTop": 70,
      },
    ]
  }
>
  <ActivityIndicator
    size="large"
  />
</View>
`;

exports[`renders correctly gettingData: true issueList(issues: [{issue}]) 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "marginTop": 70,
      },
    ]
  }
>
  <ActivityIndicator
    size="large"
  />
</View>
`;

exports[`renders correctly issues list item icon 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#438440",
        "fontSize": 20,
      },
      undefined,
      Object {
        "fontFamily": "FontAwesome",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`renders correctly issues list item icon with state: closed 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#9a76ef",
        "fontSize": 20,
      },
      undefined,
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`renders correctly issues list item icon with state: closed, pull request 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#f24141",
        "fontSize": 20,
      },
      undefined,
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`renders correctly issues list item icon with state: closed, pull request merged at 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#9a76ef",
        "fontSize": 20,
      },
      undefined,
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`renders correctly issues list item icon with state: open 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#438440",
        "fontSize": 20,
      },
      undefined,
      Object {
        "fontFamily": "FontAwesome",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`renders correctly issues list item icon with state: open, draft: true 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#94988e",
        "fontSize": 20,
      },
      undefined,
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`renders correctly issues list item icon with state: open, pull request 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#438440",
        "fontSize": 20,
      },
      undefined,
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;
